akka-docs/src/main/paradox/scala/dispatchers.md

```diff
@@ -140,3 +163,254 @@ Note that it's not guaranteed that the *same* thread is used over time, since th
+
+Java
+:   @@snip [BlockingDispatcherSample.java]($akka$/akka-docs/src/test/java/jdocs/actor/BlockingDispatcherTest.java) { #blocking-main }
+
+
+Here the app is sending 100 messages to `BlockingFutureActor` and `PrintActor` and large numbers
+of `akka.actor.default-dispatcher` threads are handling requests. When you run run the above code,
+you will likely to see the entire application gets stuck somewhere like this:
+
+```
+>　PrintActor: 44
+>　PrintActor: 45
+```
+
+`PrintActor` is considered non-blocking, however it is not able to proceed with handling the remaining messages,
+since all the threads are occupied and blocked by the other blocking actor - thus leading to thread starvation.
+
+In the thread state diagrams below the colours have the following meaning:
+
+ * Turquoise - Sleeping state
+ * Orange - Waiting state
+ * Green - Runnable state
+
+The thread information was recorded using the YourKit profiler, however any good JVM profiler
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder).
+
+The orange portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amount of turquoise (blocked, or sleeping as in our example) threads
+is very bad and leads to thread starvation.
+
+@@@ note
+
+If you own a Lightbend subscription you can use the commercial [Thread Starvation Detector](http://developer.lightbend.com/docs/akka-commercial-addons/current/starvation-detector.html)
+which will issue warning log statements if it detects any of your dispatchers suffering from starvation and other.
+It is a helpful first step to identify the problem is occurring in a production system,
+and then you can apply the proposed solutions as explained below.
+
+@@@
+
+![dispatcher-behaviour-on-bad-code.png](../images/dispatcher-behaviour-on-bad-code.png)
+
+In the above example we put the code under load by sending hundreds of messages to the blocking actor
+which causes threads of the default dispatcher to be blocked.
+The fork join pool based dispatcher in Akka then attempts to compensate for this blocking by adding more threads to the pool
+(`default-akka.actor.default-dispatcher 18,19,20,...`).
+This however is not able to help if those too will immediately get blocked,
+and eventually the blocking operations will dominate the entire dispatcher.
+
+In essence, the `Thread.sleep` operation has dominated all threads and caused anything
+executing on the default dispatcher to starve for resources (including any actor
+that you have not configured an explicit dispatcher for).
+
+@@@ note
+
+If you wrap up the blocking operation inside `scala.conccurrent.blocking` as follows:
+
+```scala
+Future {
+  scala.concurrent.blocking{
+    Thread.sleep(5000) //block for 5 seconds
+    println(s"Blocking future finished ${i}")
+  }
+}
+```
```

Comment:
ok, I don't know. Might be something with how the FJP tasks are wrapped.
If we want to show something for an actor we could use `ask` + `Await` to illustrate the `blocking`.
However, we should perhaps not go too deep into that `scala.conccurrent.blocking`.